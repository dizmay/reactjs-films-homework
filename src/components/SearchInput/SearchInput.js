import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styles from './SearchInput.scss';

const SearchInput = ({ search }) => {
  const [value, setValue] = useState('');
  const onSearchChange = (e) => {
    setValue(e.currentTarget.value);
  };
  const startSearch = (e) => {
    e.preventDefault();
    search(value);
  };
  return (
    <form id="search" className={styles.search} onSubmit={startSearch}>
      <input type="search" value={value} onChange={onSearchChange} form="search" className={styles.search__input} placeholder="Search..." />
      <button type="submit" className={styles.search__btn}>âŒ•</button>
    </form>
  );
};

SearchInput.defaultProps = {
  search: null,
};

SearchInput.propTypes = {
  search: PropTypes.func,
};

export default SearchInput;
