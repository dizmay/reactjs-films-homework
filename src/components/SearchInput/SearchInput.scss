.search {
  display: flex;
  align-items: center;
  border: 0.2rem solid #fff;
  border-radius: 0.3rem;
  height: 2.5rem;
  width: 20rem;
}

.search__input {
  border: none;
  outline: none;
  background: transparent;
  width: 17.5rem;
  height: 100%;
  padding: 0.1rem 0.2rem;
  font-size: 1rem;
  color: #fff;
}

.search__input::placeholder {
  color: #fff;
  font-size: 1rem;
}

.search__btn {
  width: 1.5rem;
  height: 1.5rem;
  color: #fff;
  margin: 1rem;
  cursor: pointer;
}
