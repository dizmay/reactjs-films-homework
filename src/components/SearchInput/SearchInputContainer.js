import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchFound } from '../../modules/Search/searchActions';
import SearchInput from './SearchInput';

const SearchInputContainer = ({ search }) => <SearchInput search={search} />;

SearchInputContainer.defaultProps = {
  search: null,
};

SearchInputContainer.propTypes = {
  search: PropTypes.func,
};

const mapDispatchToProps = (dispatch) => ({ search: (query) => dispatch(fetchFound(query)) });

export default connect(null, mapDispatchToProps)(SearchInputContainer);
