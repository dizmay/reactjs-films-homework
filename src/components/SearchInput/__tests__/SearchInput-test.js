import React from 'react';
import renderer from 'react-test-renderer';
import SearchInput from '../SearchInput';

it('renders correctly', () => {
  const tree = renderer.create(<SearchInput />).toJSON();
  expect(tree).toMatchSnapshot();
});
