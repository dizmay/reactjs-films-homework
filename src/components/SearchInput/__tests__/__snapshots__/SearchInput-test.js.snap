// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="search"
  id="search"
  onSubmit={[Function]}
>
  <input
    className="search__input"
    form="search"
    onChange={[Function]}
    placeholder="Search..."
    type="search"
    value=""
  />
  <button
    className="search__btn"
    type="submit"
  >
    âŒ•
  </button>
</form>
`;
